packages/flutter/lib/src/material/tab_controller.dart

```diff
@@ -313,7 +313,7 @@ class DefaultTabController extends StatefulWidget {
     @required this.child,
   }) : assert(initialIndex != null),
        assert(length >= 0),
-       assert(initialIndex >= 0 && initialIndex < length),
+       assert(initialIndex >= 0 && initialIndex <= length),
```

Comment:
I'm unsure if this is actually a safe change, since for all other cases, `initialIndex` can now be set to a value outside of the range of tab values (e.g. Having two tabs and having `initialIndex` set to 2 will no longer trigger this assert), This will probably be better:
```suggestion
       assert(length == 0 || (initialIndex >= 0 && initialIndex <= length)),
```

Reply:
Yes! Good point.

Reply:
@shihaohong Thanks for the feedback! If I've understood your point correctly, the code should be changed to this instead?

```suggestion
              assert(length == 0 || (initialIndex >= 0 && initialIndex < length)),
```

Reply:
Yes